### PR TITLE
Get Filesystem.SYSTEM working on Android 15

### PR DESCRIPTION
**Pull request by swankjesse**
There is no java.nio.file until API 26 (Android 8.0).

### CODE REVIEW

File: android-test/build.gradle [hunk 1]
```diff
@@ -15,8 +15,9 @@ def isIDE = properties.containsKey('android.injected.invoked.from.ide') ||
 
 android {
   compileOptions {
-    sourceCompatibility JavaVersion.VERSION_1_7
-    targetCompatibility JavaVersion.VERSION_1_7
+    sourceCompatibility JavaVersion.VERSION_1_8
+    targetCompatibility JavaVersion.VERSION_1_8
+    coreLibraryDesugaringEnabled true
```
Comment (swankjesse) [inline android-test/build.gradle:20]: Required by kotlinx.time!

File: android-test/build.gradle [hunk 2]
```diff
@@ -31,6 +32,12 @@ android {
     versionCode 1
     versionName "1.0"
     testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+
+    // AndroidJUnitRunner wasn't finding tests in multidex artifacts when running on Android 4.0.3.
+    // Work around by adding all Okio classes to the keep list. That way they'll be in the main
+    // .dx file where TestRequestBuilder will find them.
+    multiDexEnabled true
+    multiDexKeepProguard file('multidex-config.pro')
```
Comment (swankjesse) [inline android-test/build.gradle:40]: Getting here was a long and frustrating debug session.

Comment (swankjesse) [reply android-test/build.gradle:40]: (We don’t exceed any 65K limits, but coreLibraryDesugaring needs multidex)

File: docs/index.md [hunk 3]
```diff
@@ -78,6 +78,15 @@ works and how Okio does it.
 Okio’s implementation language from Java to Kotlin.
 
 
+Requirements
+------------
+
+Okio supports Android 4.0.3+ (API level 15+) and Java 7+.
```
Comment (swankjesse) [inline docs/index.md:84]: Previously our minimums weren’t specified! Note that this is lower than OkHttp’s minimums.

**Review by Egorand — APPROVED** [review]

File: android-test/build.gradle (hunk 2 above)
Comment (swankjesse) [reply android-test/build.gradle:40]: I’m treating Android 4.0.3 as the lowest possible version any reasonable application would target. I’d like, for example, to have current versions of Okio 2.x working underneath OkHttp 3.12.x for example.

File: docs/index.md (hunk 3 above)
Comment (swankjesse) [reply docs/index.md:84]: I’m assuming 4.0.3 is the lowest version that anyone is targeting for anything.  Should we go lower?

File: build.gradle [hunk 4]
```diff
@@ -22,7 +22,8 @@ buildscript {
 
   ext.deps = [
       'android': [
-        "gradlePlugin": "com.android.tools.build:gradle:4.0.2",
+        'gradlePlugin': "com.android.tools.build:gradle:4.0.2",
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply build.gradle:25]: Fixed.

File: build.gradle [hunk 5]
```diff
@@ -22,7 +22,8 @@ buildscript {
...
+        'desugarJdkLibs': "com.android.tools:desugar_jdk_libs:1.1.1",
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply build.gradle:26]: Without it I get this:

```
A problem occurred configuring project ':android-test'.
> coreLibraryDesugaring configuration contains no dependencies. If you intend to enable core library desugaring, please add dependencies to coreLibraryDesugaring configuration.
```

File: docs/index.md [hunk 6]
```diff
@@ -78,6 +78,15 @@ works and how Okio does it.
...
+
+Okio depends the [Kotlin standard library][kotlin]. It is a small library with strong
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply docs/index.md:86]: Fixed!

File: okio/src/jvmMain/kotlin/okio/Platform.kt [hunk 7]
```diff
@@ -19,7 +19,14 @@ import okio.Path.Companion.toPath
 import java.io.File
 
 @ExperimentalFilesystem
-internal actual val PLATFORM_FILESYSTEM: Filesystem = JvmSystemFilesystem
+internal actual val PLATFORM_FILESYSTEM: Filesystem = run {
```
[the remark this answers is not in the document]
Comment (swankjesse) [reply okio/src/jvmMain/kotlin/okio/Platform.kt:22]: I think it reads better than a naked `try` expression? I might be overthinking it.

File: docs/index.md (hunk 3 above)
Comment (Egorand) [reply docs/index.md:84]: No, I don't think so

File: build.gradle (hunk 5 above)
[the remark this answers is not in the document]
Comment (Egorand) [reply build.gradle:26]: Interesting, @JakeWharton would know what's going on

File: okio/src/jvmMain/kotlin/okio/Platform.kt (hunk 7 above)
[the remark this answers is not in the document]
Comment (Egorand) [reply okio/src/jvmMain/kotlin/okio/Platform.kt:22]: Not a big deal anyway

**Pull request merged**